rust/main/agents/relayer/src/server/message_retry.rs

```diff
@@ -137,25 +263,52 @@ mod tests {
             }
         ]);
 
+        // spawn a task to respond to message retry request
+        let task = async move {
+            if let Ok(req) = retry_req_rx.recv().await {
+                // Check that the list received by the server matches the pending operation
+                assert!(req
+                    .pattern
+                    .op_matches(&(Box::new(pending_operation.clone()) as QueueOperation)));
+                let resp = MessageRetryResponse {
+                    uuid: req.uuid,
+                    evaluated: 10,
+                    matched: 2,
+                };
+                req.transmitter.send(resp).await.unwrap();
+            }
+        };
```

Comment:
sweet, I think you can refactor this into a function that's reused across unit tests

Reply:
Good idea :+1: